src/sentry/api/endpoints/organization_details.py

```diff
@@ -380,6 +393,16 @@ def save(self):
             org.name = data["name"]
         if "slug" in data:
             org.slug = data["slug"]
+        if "providerName" in data and "providerConfig" in data:
+            provider_name = data["providerName"]
+            provider_config = data["providerConfig"]
+            auth_provider = AuthProvider.objects.update_or_create(
```

Comment:
Is there a chance that that this AuthProvider will be created/updated but then the Organization create/update/save would fail? 

Do we need to wrap everything in a [transaction](https://docs.djangoproject.com/en/4.2/topics/db/transactions/)?

Reply:
I think to be safe I'll wrap this inside a transaction using `with transaction.atomic():`